static/js/containers/VideoDetailPage.js

```diff
@@ -0,0 +1,139 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import type { Dispatch } from "redux";
+import moment from 'moment';
+
+import Button from '../components/material/Button';
+import Dialog from '../components/material/Dialog';
+import Toolbar from '../components/material/Toolbar';
+import Footer from '../components/Footer';
+import VideoPlayer from '../components/VideoPlayer';
+import { actions } from '../actions';
+import {
+  setDialogVisibility,
+  setTitle,
+  clear,
+} from '../actions/videoDetailUi';
+import { makeCollectionUrl } from '../lib/urls';
+import type { Video } from "../flow/videoTypes";
+import type { VideoDetailUIState } from "../reducers/videoDetailUi";
+import { MM_DD_YYYY } from '../constants';
+import Textfield from "../components/material/Textfield";
+
+class VideoDetailPage extends React.Component {
+  props: {
+    dispatch: Dispatch,
+    video: ?Video,
+    videoKey: string,
+    needsUpdate: boolean,
+    videoDetailUi: VideoDetailUIState,
+  };
+
+  componentDidMount() {
+    this.updateRequirements();
+  }
+
+  componentDidUpdate() {
+    this.updateRequirements();
+  }
+
+  updateRequirements = () => {
+    const { dispatch, videoKey, needsUpdate } = this.props;
+
+    if (needsUpdate) {
+      dispatch(actions.videos.get(videoKey));
+    }
+  };
+
+  openDialog = () => {
+    const { dispatch } = this.props;
+    dispatch(setDialogVisibility(true));
+  };
+
+  submitForm = () => {
+    const { dispatch, videoKey, videoDetailUi } = this.props;
+    const { title } = videoDetailUi.dialog;
+
+    dispatch(clear());
+    dispatch(actions.videos.patch(videoKey, {
+      title,
+    }));
+  };
+
+  clearDialog = () => {
+    const { dispatch } = this.props;
+    dispatch(clear());
+  };
+
+  setTitle = (event: Event) => {
+    const { dispatch } = this.props;
+    dispatch(setTitle(event.target.value));
+  };
+
+  render() {
+    const { video, videoDetailUi } = this.props;
+    if (!video) {
+      return null;
+    }
+
+    const formattedCreation = moment(video.created_at).format(MM_DD_YYYY);
+    const collectionUrl = makeCollectionUrl(video.collection_key);
+    return <div>
+      <Toolbar />
+      <VideoPlayer
+        video={video}
+      />
+      <div className="summary">
+        <a className="collection-link" href={collectionUrl}>
+          {video.collection_title}
+        </a>
+        <span className="video-title">
+          {video.title}
```

Comment:
Video description and transcoding status used to be displayed on this page, I think it would be good to continue showing those to the user, with both title and description editable.